### ISSUE TITLE
Something is off with caching

### ISSUE THREAD

**Issue by artem-zinnatullin** [buildfoundation/bazel_rules_detekt#56]
We have following effective bazelrc config on CI (kept only relevant options):

```
build --remote_cache=somerealcacheserver --remote_upload_local_results=true
```

Yet we're seeing almost no cache hits for rules_detekt actions.
Our Buck detekt builds with exact same `srcs` and detekt config are getting up to 100% cache hits.

Other types of Bazel actions are getting proper cache hits in our setup, I'm trying to figure out where the problem is coming from but so far can't tell so I'm filing this issue.

**Comment by artem-zinnatullin** [comment]
Hm, I see "Compiling Kotlin to JVM @rules_detekt//detekt/wrapper:lib { kt: 9, java: 0, srcjars: 0 } [for host] " not getting cache hits

Might be due to rules_jvm_external or rules_kotlin (shouldn't be)

**Comment by arturdryomov** [comment]
FYI — there is [a cache debugging guide](https://docs.bazel.build/versions/master/remote-caching-debug.html).

**Comment by arturdryomov** [comment]
All right, so I’ve configured [the Nginx server](https://docs.bazel.build/versions/master/remote-caching.html#nginx). Then created a `.bazelrc` file:

```
build --remote_cache=http://localhost:8080/cache/
```

Runs:

```console
$ bazel clean
INFO: Invocation ID: 2e7eed1a-b720-42ac-b88e-64a48b725bd9
INFO: Starting clean.

$ bazel build //tests/integration:detekt_xml_report
INFO: Invocation ID: 2405a2c1-ed44-4e3b-af04-3496d5db91f1
INFO: Analyzed target //tests/integration:detekt_xml_report (94 packages loaded, 1233 targets configured).
INFO: Found 1 target...
INFO: From Linking external/com_google_protobuf/libprotobuf.a [for host]:
/Library/Developer/CommandLineTools/usr/bin/libtool: file: bazel-out/host/bin/external/com_google_protobuf/_objs/protobuf/error_listener.o has no symbols
INFO: From Linking external/com_google_protobuf/libprotobuf_lite.a [for host]:
/Library/Developer/CommandLineTools/usr/bin/libtool: file: bazel-out/host/bin/external/com_google_protobuf/_objs/protobuf_lite/io_win32.o has no symbols
INFO: From Building external/com_google_protobuf/libprotobuf_java.jar (122 source files, 1 source jar) [for host]:
warning: -parameters is not supported for target value 1.7. Use 1.8 or later.
Target //tests/integration:detekt_xml_report up-to-date:
  bazel-bin/tests/integration/detekt_xml_report_detekt_report.txt
  bazel-bin/tests/integration/detekt_xml_report_detekt_report.xml
INFO: Elapsed time: 235.775s, Critical Path: 78.18s
INFO: 219 processes: 211 darwin-sandbox, 8 worker.
INFO: Build completed successfully, 245 total actions

$ bazel clean
INFO: Invocation ID: 04bce235-bcc6-4889-a87c-3d2919013442
INFO: Starting clean.

$ bazel build //tests/integration:detekt_xml_report
INFO: Invocation ID: 555cf3f8-d1be-42e3-9e69-7736a49c89cd
INFO: Analyzed target //tests/integration:detekt_xml_report (94 packages loaded, 1233 targets configured).
INFO: Found 1 target...
INFO: From Linking external/com_google_protobuf/libprotobuf_lite.a [for host]:
/Library/Developer/CommandLineTools/usr/bin/libtool: file: bazel-out/host/bin/external/com_google_protobuf/_objs/protobuf_lite/io_win32.o has no symbols
INFO: From Linking external/com_google_protobuf/libprotobuf.a [for host]:
/Library/Developer/CommandLineTools/usr/bin/libtool: file: bazel-out/host/bin/external/com_google_protobuf/_objs/protobuf/error_listener.o has no symbols
INFO: From Building external/com_google_protobuf/libprotobuf_java.jar (122 source files, 1 source jar) [for host]:
warning: -parameters is not supported for target value 1.7. Use 1.8 or later.
Target //tests/integration:detekt_xml_report up-to-date:
  bazel-bin/tests/integration/detekt_xml_report_detekt_report.txt
  bazel-bin/tests/integration/detekt_xml_report_detekt_report.xml
INFO: Elapsed time: 3.553s, Critical Path: 1.42s
INFO: 219 processes: 219 remote cache hit.
INFO: Build completed successfully, 245 total actions
```

TL;DR — the remote cache works. I’ll attempt to reproduce in a dedicated workspace. @artem-zinnatullin, please attempt to reproduce the issue in an isolated environment if possible.

**Comment by arturdryomov** [comment]
Same thing with a dedicated workspace.

**Comment by arturdryomov** [comment]
Unable to reproduce in isolated environment, so closing this. Might have been an internal caching server issue since the current approach seem to cache results fine.
